sqlmesh/core/plan/definition.py

```diff
@@ -532,7 +535,9 @@ def _categorize_snapshots(self) -> None:
         returns a list of added and directly modified snapshots as well as the mapping of
         indirectly modified snapshots.
         """
-        for model_name, snapshot in self.context_diff.snapshots.items():
+        for model_name, snapshot in self.snapshot_mapping.items():
+            if snapshot.intervals:
```

Comment:
Btw, the subsequent checks already take care of this:
```
if model_name in self.context_diff.modified_snapshots:
    ...
elif model_name in self.context_diff.added and self.is_new_snapshot(snapshot):
    ...
```
Why have it? I might only suggest to move `upstream_model_names = self._dag.upstream(model_name)` under the 1st branch.

Reply:
> Why is this check needed here? Wouldn't it make more sense to check whether the category is set? Otherwise the check looks rather arbitrary in this context.

Yeah I agree this check can be better explained. I'm good with checking the category but I actually just matched what we do here: https://github.com/TobikoData/sqlmesh/blob/b4dbb4587edaf7eb78103479c449d915542f0a51/sqlmesh/core/snapshot/definition.py#L737

Reply:
> Why have it? 

If I don't have this and we set plan start then I hit that assert line linked above and error out. So I just did this just upfront as you can see here.